tooling/bundler/src/bundle/linux/pacman.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use super::{
+  super::common,
+  debian::{copy_custom_files, generate_data, tar_and_gzip_dir},
+};
+use crate::Settings;
+use anyhow::Context;
+use flate2::Compression;
+use heck::AsKebabCase;
+use log::info;
+use sha2::{Digest, Sha512};
+
+use std::{
+  fs::{self, File},
+  io::{self, Write},
+  path::{Path, PathBuf},
+};
+
+/// Bundles the project.
+/// Returns a vector of PathBuf that shows where the archive.tar.gz was created.
+pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
+  let arch = match settings.binary_arch() {
+    // Arch Linux recognises `x86_64` and `aarch64` instead of `amd64` and `arm64` respectively.
+    "x86" => "i386",
+    // ARM64 is detected differently, armel isn't supported, so armhf is the only reasonable choice here.
+    "arm" => "armhf",
+    other => other,
+  };
+  let package_base_name = format!(
+    "{}_{}_{}",
+    settings.main_binary_name(),
+    settings.version_string(),
+    arch
+  );
+
+  let base_dir = settings.project_out_directory().join("bundle/pacman");
+
+  let package_dir = base_dir.join(&package_base_name);
+  if package_dir.exists() {
+    fs::remove_dir_all(&package_dir)
+      .with_context(|| format!("Failed to remove old {}", package_base_name))?;
+  }
+
+  let package_name = format!("{}.tar.gz", package_base_name);
+  let package_path = base_dir.join(&package_name);
+  info!(action = "Bundling"; "{} ({})", package_name, package_path.display());
+
+  let (data_dir, _) = generate_data(settings, &package_dir)
+    .with_context(|| "Failed to build data folders and files")?;
+  copy_custom_files(settings, &data_dir).with_context(|| "Failed to copy custom files")?;
```

Comment:
this will use `files` under `deb` config, we should add a `files` option for `pacman` instead and change `copy_custom_files` function to be more generic, we should do the same for `appimage` as well

Reply:
agreed!

Reply:
Sure